tests/test_ui.py

```diff
@@ -0,0 +1,91 @@
+
+    def test_ingest(self) -> None:
+        self.ct.ingest(sc)
+        d = self.ct.to_dict()
+        assert d.get(
+            "84378aea-b64c-49a3-9539-f854532279ee").get("type") == "header"
+
+    def test_descendents(self) -> None:
+        desc = self.ct.get_descendents("root")
+        desc_ids = list(map(lambda x: x.id, desc))
+        assert "84378aea-b64c-49a3-9539-f854532279ee" in desc_ids
+        assert "bb4d0e86-619e-4367-a180-be28ab6059f4" in desc_ids
+        assert "85120b55-69c6-4b50-853a-bbbf73ff8121" in desc_ids
+
+
+class TestUIManager:
+    ui_manager = StreamsyncUIManager(session.session_id)
+
+    def test_find_component(self):
+        # Verify that the find method correctly retrieves a component by its ID
+        expected_id = "84378aea-b64c-49a3-9539-f854532279ee"
+        found_component = self.ui_manager.find(expected_id)
+        assert found_component is not None
+        assert found_component.id == expected_id
+
+    def test_assert_in_container_context(self):
+        # Verify the context assertion for creating components within a container
+        # First, successfully create a component within a container context
+        with self.ui_manager.root:
+            passed = False
+            try:
+                self.ui_manager.create_component('text', content={'text': 'Inside container'})
+                passed = True
+            except Exception:
+                passed = False
+            assert passed, "Component should be created within container context without errors"
+
+        # Now, attempt to create a component outside of any container context, which should raise an error
+        raised = False
+        try:
+            self.ui_manager.create_component('text', content={'text': 'Outside container'})
+        except UIError:
+            raised = True
+        assert raised, "Creating a component outside of a container context should raise an UIError"
+
+    def test_column_container_creation(self):
+        container = self.ui_manager.ColumnContainer(id="test_container")
+        assert container.type == 'columns'
+        assert container.id == "test_container"
+        assert container in self.ui_manager.session.session_component_tree.components.values()
+
+    def test_column_container_creation_no_id(self):
+        container = self.ui_manager.ColumnContainer()
+        assert container.type == 'columns'
+        assert container.id is not None
+        assert container in self.ui_manager.session.session_component_tree.components.values()
+        assert container.id in self.ui_manager.session.session_component_tree.components.keys()
+
+    def test_column_creation(self):
+        column = self.ui_manager.Column(id="test_column")
+        assert column.type == 'column'
+        assert column.id == "test_column"
+        assert column in self.ui_manager.session.session_component_tree.components.values()
+
+    def test_text_creation(self):
+        with self.ui_manager.root:
```

Comment:
I think `root` accept usually only pages in streamsync. I don't think it's a good way to start playing with user interface. What's happen when you set this in the json file ?